lib/vm/src/cache.rs

```diff
@@ -120,8 +120,7 @@ where
     pub fn store_instance(&mut self, id: &[u8], instance: Instance<S, A>) -> Option<Extern<S, A>> {
         if let Some(cache) = &mut self.instances {
             let hash = WasmHash::generate(&id);
-            let storage = instance.take_storage();
-            let (wasmer_instance, api) = Instance::recycle(instance);
+            let (wasmer_instance, api, storage) = Instance::recycle(instance);
```

Comment:
Okay, you take all the pieces.

`Extern` will expand soon (with querier interface), and i figured nicest is simply to return `Extern`, as that will always contain everything we store on the instance